### ISSUE TITLE
batch size question

### ISSUE THREAD

**Issue by jessejchuang** [2017-fall-DL-training-program/PyTorchWarmUp#2]
I quote one paragraph from Goodfellow's textbook: "The minibatch size m' is typically chosen to be a relatively small number of examples, **ranging from 1 to a few hundred**. Crucially, **m' is usually held fixed** as the training set size m grows. We may fit a training set with billions of examples **using updates computed on only a hundred examples**.".

I remember one TA said last Wed that one Google paper is using 1024 batch size. For MNIST, 1024 batch needs 8 times epochs to get the same iterations as 128 batch. That takes time to do training. Moreover, the 1024 batch with 160 epochs performs poor than 128 batch with 20 epochs. I'm wondering if Google's 1024 batch size is a special case. **We are supposed to keep using hundreds of batch size as suggested by Goodfellow in the following Labs and final project. Any suggestion?**

**Comment by JiaRenChang** [comment]
Hi,
   We recommend you this paper "[Accurate, Large Minibatch SGD: Training ImageNet in 1 Hour](https://arxiv.org/pdf/1706.02677.pdf)".
   This paper is proposed by Facebook AI Research.
   They compared different size of batch.
   You can see the effect of batch size in Figure 1.
   By the way, we will offer hyper-parameters for training in the following Labs.

**Comment by jessejchuang** [comment]
Hi TA,
Thanks for sharing the paper. I have no much time and just take a look at Abstract and figure 1. I quote one paragraph here, "Specifically, we show no loss of accuracy when training with large minibatch sizes up to 8192 images. To achieve this result, we **adopt a linear scaling rule for adjusting learning rates as a function of minibatch size** and **develop a new warmup scheme that overcomes optimization challenges early in training**."

It seems larger batch size should accompany some kind of adjusting learning rate and warmup scheme. Finally, thanks for offering hyper-parameters of Labs.
